ic-http/src/transform.rs

```diff
@@ -83,4 +84,48 @@ mod test {
         // Assert.
         assert_eq!(names, vec!["transform_function_1", "transform_function_2"]);
     }
+
+    /// Transform function which is intentionally creates a new request
+    /// with itself as a transform function.
+    /// This is potentially causing to rewrite the transform function
+    /// in a thread-local storage while it is being executed, which
+    /// leads to a hang.
+    fn transform_function_with_overwrite(arg: TransformArgs) -> HttpResponse {
+        create_request_with_transform();
+        arg.response
+    }
+
+    /// Creates a request with a transform function which overwrites itself.
+    fn create_request_with_transform() -> CanisterHttpRequestArgument {
+        crate::request::create_request()
+            .url("https://www.example.com")
+            .transform_func(transform_function_with_overwrite, vec![])
+            .build()
+    }
```

Comment:
Actually, reading this function, how about we move `args.response` from above inside here directly? Not sure we gain much by having the extra `transform_function_with_overwrite`.

Reply:
This has to be a separate function with a name so it can be stored in a thread-local storage by it's name.